RFS/src/main/java/com/rfs/tracing/RfsContexts.java

```diff
@@ -0,0 +1,453 @@
+            meterIncrementEvent(getMetrics().bytesReadCounter, i, getSharedAttributes(Attributes.builder()));
+        }
+    }
+
+    public static class CheckedIdempotentPutRequestContext
+            extends BaseSpanContext<RootRfsContext>
+            implements IRfsContexts.ICheckedIdempotentPutRequestContext {
+        @Getter
+        public final IScopedInstrumentationAttributes enclosingScope;
+        private final String label;
+
+        public CheckedIdempotentPutRequestContext(RootRfsContext rootScope,
+                                                  IScopedInstrumentationAttributes enclosingScope,
+                                                  String label) {
+            super(rootScope);
+            initializeSpan(rootScope);
+            this.enclosingScope = enclosingScope;
+            this.label = label;
+        }
+
+        @Override
+        public String getActivityName() { return ACTIVITY_NAME; }
+
+        public static class MetricInstruments extends CommonScopedMetricInstruments {
+            private MetricInstruments(Meter meter, String activityName) {
+                super(meter, activityName);
+            }
+        }
+
+        public static @NonNull MetricInstruments makeMetrics(Meter meter) {
+            return new MetricInstruments(meter, ACTIVITY_NAME);
+        }
+
+        // If we want separate metrics, we could key off of additional attributes like uri/verb to get values
+        // from maps within the root context to retrieve metric objects that are setup at runtime rather
+        // than compile time
+        @Override
+        public MetricInstruments getMetrics() {
+            return getRootInstrumentationScope().getTwoStepIdempotentRequestInstruments;
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createCheckRequestContext() {
+            return new GenericRequestContext(rootInstrumentationScope, this,
+                    label+"createCheckRequestContext");
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createPutContext() {
+            return new GenericRequestContext(rootInstrumentationScope, this,
+                    label+"createPutContext");
+        }
+
+    }
+
+    public static class CreateSnapshotContext
+            extends BaseSpanContext<RootRfsContext>
+            implements IRfsContexts.ICreateSnapshotContext {
+
+        protected CreateSnapshotContext(RootRfsContext rootScope) {
+            super(rootScope);
+            initializeSpan(rootScope);
+        }
+
+        @Override
+        public String getActivityName() { return ACTIVITY_NAME; }
+
+        @Override
+        public IScopedInstrumentationAttributes getEnclosingScope() {
+            return null;
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createRegisterRequest() {
+            return new GenericRequestContext(rootInstrumentationScope, this, "createRegisterRequest");
+        }
```

Comment:
Maybe this is for flexiblity, but the method that calls `createRegisterRequest()` is registerSnapshotRepo, seems like the name should be from the calling method annotated onto the GenericRequestContext.

What I would consider a better alternative is to have the caller provide the label, e.g.
```
createRegisterRequest(...) {
     ...
     
     try (final var requestContext = context.createRequestContext("createRegisterRequest")) {
        // Use it
     }
     ...
}
```

It would be even better if this was automatically using the callstack